src/YamlValidator/YamlLoader.cs

```diff
@@ -25,7 +25,7 @@ public class YamlLoader
                 var yamlFiles = Directory.EnumerateFiles(filePath, "*" + Constants.YamlFileExtension, SearchOption.AllDirectories);
                 foreach (var filename in yamlFiles)
                 {
-                    var fileName = Path.GetFileName(filename);
+                    var fileName = Path.GetRelativePath(filePath, filename);
```

Comment:
I'm not 100% sure, but I think https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfullpath?view=net-8.0#system-io-path-getfullpath(system-string-system-string) would be a better choice

Reply:
My thinking in using relative paths was that the users wouldn't have to deal with verbose file paths (they input the folder path so the relative paths should be straightforward). In the CI, this would mean we wouldn't get an uninformative longer path from the container's absolute file path.

Reply:
I think using the full path is much safer though so I guess we can go with that

Reply:
Can you elaborate? As in use a full directory path? 
> Using a predefined basePath rather than one based on the current drive directory guards against unwanted file paths caused by unexpected changes in the current drive and directory. (from the docs)
We would naturally expect the user to not change the location of the path they input when running the cli tool

Reply:
correct. I'd suggest using the complete absolute path to the file as dictionary key.